buildSrc/src/main/java/gradlebuild/JniNokeePlugin.java

```diff
@@ -45,73 +70,111 @@ public void apply(Project project) {
-                } else {
-                    return ImmutableList.of(); // do nothing
+    private void configureVariants(Project project) {
+        library(project, library -> {
+            VariantsExtension variants = project.getExtensions().getByType(VariantsExtension.class);
+            variants.getVariantNames().set(library.getVariants().flatMap(new Transformer<Iterable<String>, JniLibrary>() {
+                @Override
+                public Iterable<String> transform(JniLibrary it) {
+                    // Only depend on variants which can be built on the current machine
+                    boolean onlyLocalVariants = project.getProviders().gradleProperty("onlyLocalVariants").forUseAtConfigurationTime().isPresent();
+                    if (onlyLocalVariants && !isBuildable(it)) {
+                        return ImmutableList.of();
+                    } else {
+                        return ImmutableList.of(JniNokeePlugin.VariantNamer.INSTANCE.determineName(it));
+                    }
                 }
-            }));
-        });
-        library.getTasks().configureEach(LinkSharedLibrary.class, task -> {
-            task.getLinkerArgs().addAll(task.getTargetPlatform().map(targetPlatform -> {
-                OperatingSystem targetOs = targetPlatform.getOperatingSystem();
-                if (targetOs.isMacOsX()) {
-                    return ImmutableList.of(
-                        "-mmacosx-version-min=10.9",
-                        "-framework", "CoreServices");
-                } else if (targetOs.isWindows()) {
-                    return ImmutableList.of("Shlwapi.lib", "Advapi32.lib");
-                } else {
-                    return ImmutableList.of(); // do nothing
+
+                private boolean isBuildable(JniLibrary variant) {
+                    if (!variant.getSharedLibrary().isBuildable()) {
+                        return false; // strait-up not buildable
+                    } else {
+                        @SuppressWarnings("unchecked")
+                        final Provider<Set<NcursesVersion>> availableNcursesVersions = (Provider<Set<NcursesVersion>>) project.getExtensions().findByName("availableNcursesVersions");
+                        if (availableNcursesVersions == null) {
+                            return true; // no known ncurses versions, assuming buildable
+                        } else {
+                            // For each variant with the ncurses dimension, check if the version is available
+                            if (variant.getBuildVariant().hasAxisOf(NCURSES_5)) {
+                                return availableNcursesVersions.get().contains(NCURSES_5);
+                            } else if (variant.getBuildVariant().hasAxisOf(NCURSES_6)) {
+                                return availableNcursesVersions.get().contains(NCURSES_6);
+                            }
+                            return true;
+                        }
```

Comment:
🔍 Nokee doesn't support the ad-hoc buildable flag, instead we need to verify the buildability manually for custom dimensions. In the future, we will move toward including the compile/link dependencies to verify the buildability. If the dependencies do not resolve successfully, we can assert the variant is not buildable. It opens the unfortunate possibility that users force dependencies to resolves too early, however, early resolution of provider check should take care of this issue.